isthmus/src/test/java/io/substrait/isthmus/SubstraitRelNodeConverterTest.java

```diff
@@ -236,4 +243,58 @@ public void emit() {
       assertRowMatch(relNode.getRowType(), R.I32, N.STRING);
     }
   }
+
+  @Nested
+  class AdvancedExtensions {
+
+    AdvancedExtension advancedExtension =
+        AdvancedExtension.builder()
+            .optimization(new EmptyOptimization())
+            .enhancement(new EmptyEnhancement())
+            .build();
+
+    class CustomSubstraitRelNodeConverter extends SubstraitRelNodeConverter {
+
+      public CustomSubstraitRelNodeConverter(
+          SimpleExtension.ExtensionCollection extensions,
+          RelDataTypeFactory typeFactory,
+          RelBuilder relBuilder) {
+        super(extensions, typeFactory, relBuilder);
+      }
+
+      @Override
+      public RelNode visit(io.substrait.relation.NamedScan namedScan) throws RuntimeException {
+        RelNode relNode = super.visit(namedScan);
+
+        // optimization and enhancements that are set explictily are propagated
+        assertInstanceOf(
+            EmptyEnhancement.class, namedScan.getExtension().get().getEnhancement().get());
+        assertInstanceOf(
+            EmptyOptimization.class, namedScan.getExtension().get().getOptimization().get());
+
+        return relNode;
+      }
+    }
+
+    class CustomSubstraitToCalcite extends SubstraitToCalcite {
+
+      public CustomSubstraitToCalcite(
+          SimpleExtension.ExtensionCollection extensions, RelDataTypeFactory typeFactory) {
+        super(extensions, typeFactory);
+      }
+
+      @Override
+      protected SubstraitRelNodeConverter createSubstraitRelNodeConverter(RelBuilder relBuilder) {
+        return new CustomSubstraitRelNodeConverter(extensions, typeFactory, relBuilder);
+      }
+    }
+
+    @Test
+    public void defaultValue() {
+      Plan.Root root = b.root(b.namedScan(List.of("foo"), List.of(), List.of(), advancedExtension));
+
+      var customConverter = new CustomSubstraitToCalcite(extensions, typeFactory);
+      var relNode = customConverter.convert(root.getInput());
```

Comment:
Would it be possible to add some kind of assertion here?

Reply:
For sure. I want to add some more testing once we're aligned on an approach.